Intel Open Image Denoise integration

bamless opened the pull request:
With this pull request, we integrate OIDN in Yocto/GL in the form of a new library and app.
Quick summary of changes and additions:
 - Added a new app `yimagedenoise` that takes in a noisy path traced image (LDR or HDR), optional albedo and normal feature images (LDR or HDR for albedo, HDR for normal), and outputs the denoised result.
 - Added two new tracers, namely trace_albedo and trace_normal, to libs/yocto_trace. These are used to generate feature images needed by the intel open image denoise library. These differ from the false color 'diffuse' and 'normal' tracers in the handling of transmissions, reflections, transparency and are generally tailored to the need of OIDN.
 - Added two new values that the '-t' option of yscenetrace can accept. These are 'dalbedo' and 'dnormal' (the d stands for denoise) and they correspond respectively to the trace_albedo and trace_normal tracer. 
- Added '-d,--denoise-features' option to yscenetrace. Toggling this option results in the generation of feature images of the scene alongside the main path traced result. This option can be used to avoid the re-loading of the scene that occurs when using the -t daledo and -t dnormal options standalone.
- Added a new library `yocto_denoise`. This library wraps the functionality of Intel Open Image Denoise in a way that conforms to the design of Yocto/GL, by using Yocto types and conventions.
- To enable the compilation of OIDN and the denoise app and library, a new option has been added to the main CMakeLists.txt file: YOCTO_DENOISE. This option is set to off as default.
- To provide a convinient way of obtaining OIDN, a new git submodule to the OIDN github repo has been added in libs/yocto_denoise/ext. This has some benefits, a notable one being the non-reliance on precompiled binaries that have to be compiled for every platform Yocto/GL wishes to support. On the other hand, OIDN is a fairly big dependency both in terms of compilation time and download size, and I'm not convinced this is necessarily the best approach of including it in Yocto. Suggestions on the issue are welcome.
 - Updated README.md file with sections regarding the new denoise app and library

**Rebased denoise branch**

Rebased all the changes to 'version3' branch. I also squashed all previous commits down to 2 to provide a more cohesive commit histroy.
The version3 branch has been tested with the new additions and it does compile and produce correct outputs.

xelatihy commented:
I like this change a lot and I would like to include it. But I am about to release a new version of the library with some changes to namespaces and types. The changes are so that I cannot merge your version as is. There are two options we can take.

1. I create a branch on my repo and accept your PR in that branch --- I will then manually merge your changes to version3
2. You can rebase your change to version3, which is about to be release barring documetation

I am happy to do 1., but wanted to let you know before proceeding.

Thanks for this PR!

bamless commented:
I went ahead and rebased the denoise branch on the 'version3' one and reworked all parts that didn't compile anymore. I also re-tested it thoroughly to confirm that the denoiser additions still produce correct results.

xelatihy commented:
Sounds really good. Only a tiny amount of cleanup will be needed after merging, but it works just as well this way.
